aws/cognito_user_pool_ui_customization.go

```diff
@@ -0,0 +1,119 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func cognitoUserPoolUICustomizationSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"css": {
+					Type:     schema.TypeString,
+					Optional: true,
+					DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+						return new == stringHashSum(old)
+					},
+					StateFunc: func(v interface{}) string {
+						switch v.(type) {
```

Comment:
Looks as though some minor changes are required here to get the tests passing lints. `switch v := v.(type)` and remove the string type assertion.